app/scripts/utils/functional.es6

```diff
@@ -81,3 +108,11 @@ Result.mapObj = (fn, obj) => reduceResult(fn, (a, b) => a.concat(b), [], asArray
  * @returns {Result<object>} The merged object in a Result.Ok
  */
 Result.merge = (data) => reduceResult((v) => Ok(v), mergeTwo, {}, data);
+
+/**
+ * @param {array<object>} data Some objects to be aggregated and wrapped in Result.Ok
+ * @param {string} aggProp the property name to aggregate
+ * @param {string} countProp the property name to count
+ * @returns {Result<object>} The object in a Result.Ok
+ */
+Result.aggregate = (data, aggProp, countProp) => reduceResult(checkData, (agg, item) => sumAggregate(agg, item, aggProp, countProp), {keys: {aggProps: [], sumProps: []}}, data);
```

Comment:
and if `a` or `b` is missing on any of `data`, the whole thing is `Err('There is not data')`